tools/run_all_tests.py

```diff
@@ -132,7 +132,19 @@ def test_all(
 
     # Filter tests by extension
     if extension is not None:
-        all_test_paths = [test_path for test_path in all_test_paths if extension in test_path]
+        all_tests_in_selected_extension = []
+
+        for test_path in all_test_paths:
+            # Extract extension name from test path
+            extension_name = test_path[test_path.find("extensions") :].split("/")[1]
```

Comment:
If all this is hard-coded to the isaaclab repo, wouldn't it just be simpler to set the test directory itself through `--test_dir`? Wondering now if we even need a special extension case?

Reply:
Hm okay. Then the only other thing: Instead of skipping it here as a filtering step, would it make more sense to just add this to the `test_dir` before doing the glob? Unless there is a corner case I'm not seeing, doing this exhaustive for-loop might be slow (depends how many tests we have).

Reply:
This for loop would be minimal as far as timing, so I don't think it's worth optimizing given it's easily readable

Reply:
Fair enough. LGTM.